pcdsdevices/device.py

```diff
@@ -261,3 +263,70 @@ def to_interface(device_class):
         (InterfaceDevice, device_class),
         interface_cpts
         )
+
+
+class UpdateComponent(Component):
+    """
+    A component that copies and updates a parent component in a subclass.
+
+    Use this like any other component, adding it to your device that is a
+    subclass of another device, using the same name as the component you'd like
+    to update.
+
+    Pass to this component class the arguments you'd like to change from the
+    subdevice component definition.
+
+    Some limitations:
+
+    - It is not possible to use this to change the component class itself, e.g.
+      if you would like to change something from a "Component" to a
+      "FormattedComponent".
+    - All arguments to update must be provided as keyword arguments, so we know
+      which value to change in the component class.
+    - UpdateComponent will always fail isinstance checks for Component
+      subclasses.
+    - UpdateComponent is not mutable after instantiation.
+
+    Parameters
+    ----------
+    **kwargs: any, optional
+        keyword arguments to update
+    """
+    def __init__(self, **kwargs):
+        # Store the original kwargs for use later
+        self.update_kwargs = kwargs
+        # Begin with "something" as the copy_cpt to avoid issues on edge cases.
+        self.copy_cpt = None
+
+    def __set_name__(self, owner, attr_name):
+        # Find the parent cpt of the same name and copy it
+        parent_cpt = getattr(owner.mro()[1], attr_name)
```

Comment:
I think this is good enough for the majority of cases, but you might search the `mro` to deal with multiple inheritance.

Reply:
Ahh, good point, multiple inheritance breaks the simple grab here.

Reply:
Fixed this now by iterating through `mro()[1:]` (skip own class)